// account/index.js
/*
    methods {
        userExists(userId)
        newUser(userId, refreshToken)
        getUser(userId)
        updateRefreshToken(userId, refreshToken)
    }
*/

var fs = require('fs')
var path = require('path')

var filePath = path.join(__dirname, "../..", 'userData.json')

function readUsers() {
  if (!fs.existsSync(filePath)) 
    return {}
  var data = fs.readFileSync(filePath);
  return JSON.parse(data);
}

function writeUsers(users) {
  fs.writeFileSync(filePath, JSON.stringify(users))
}



module.exports = {
  userExists: function(userId) {
    var users = readUsers()
    var exists = users.hasOwnProperty(userId)
    return exists
  },

  newUser: function(userId, refreshToken) {
    if(!this.userExists(userId)) {
        var users = readUsers()
        users[userId] = { 
            refreshToken 
        };
        writeUsers(users)
    }
  },

  getUser: function(userId) {
    var users = readUsers()
    return users[userId] || null
  },

  updateRefreshToken: function(userId, refreshToken) {
    if (this.userExists(userId)) {
        var users = readUsers()
        users[userId].refreshToken = refreshToken
        writeUsers(users)
    }
  }
};
