// account/index.js
/*
    methods {
        userExists(userId)
        newUser(userId, refreshToken)
        getUser(userId)
        updateRefreshToken(userId, refreshToken)
    }
*/

var fs = require('fs')
var path = require('path')

var filePath = path.join(__dirname, "../..", 'userData.json')

function readUsers() {
  if (!fs.existsSync(filePath)) 
    return []
  var data = fs.readFileSync(filePath);
  return JSON.parse(data);
}

function writeUsers(users) {
  fs.writeFileSync(filePath, JSON.stringify(users, null, 2));
}



module.exports = {
  userExists: function(userId) {
    var users = readUsers()
    return users.some(function(user) {
        return user.id === userId
    })
  },

  newUser: function(userId, refreshToken) {
    if(!this.userExists(userId)) {
        var users = readUsers()
        users.push({ 
            "id": userId, 
            "refreshToken": refreshToken 
        })
        writeUsers(users)
    }
  },

  getUser: function(userId) {
    var users = readUsers()
    var user =  users.find(function(user) {
        return user.id === userId
    })
    return user || null
  },

  updateRefreshToken: function(userId, refreshToken) {
    var users = readUsers()
    for (var i = 0; i < users.length; i++) {
        if (users[i].id === userId) {
        users[i].refreshToken = refreshToken
        break
        }
    }
    writeUsers(users)
  }
};
